File: jdroid-java-core/src/test/java/com/jdroid/java/CollectionMatcher.java
package com.jdroid.java;

import org.mockito.ArgumentMatcher;

import java.util.Collection;

/**
 * Matcher that verify the argument is contained by the collection
 *
 * @param <E> The collection type
 */
public class CollectionMatcher<E> extends ArgumentMatcher<E> {

	private Collection<E> collection;

	/**
	 * @param collection The collection
	 */
	public CollectionMatcher(Collection<E> collection) {
		this.collection = collection;
	}

	@Override
	public boolean matches(Object argument) {
		return this.collection.contains(argument);
	}

}

File: jdroid-java-core/src/test/java/com/dipien/java/CollectionMatcher.kt
package com.dipien.java

import org.mockito.ArgumentMatcher

/**
 * Matcher that verify the argument is contained by the collection
 *
 * @param <E> The collection type
</E> */
class CollectionMatcher<E>(private val collection: Collection<E>) : ArgumentMatcher<E>() {

    @Suppress("UNCHECKED_CAST")
    override fun matches(argument: Any): Boolean {
        return this.collection.contains(argument as E)
    }
}
